plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/UIUtils.java

```diff
@@ -413,13 +413,41 @@ public static void setVisibleAndExclude(boolean visible, final Control control)
 	 * @param button
 	 */
 	public static void setDefaultButtonWidth(Button button) {
-		Assert.isLegal(!DisposeUtils.isDisposed(button));
-		Assert.isLegal(button.getLayoutData() instanceof GridData, "the given button is not layouted with a GridLayout");
-
+		assertCanSetGridData(button);
 		((GridData)button.getLayoutData()).widthHint =  
 				convertHorizontalDLUsToPixels(button, IDialogConstants.BUTTON_WIDTH);
 	}
 
+	/**
+	 * Sets equal width to all buttons in the array, which is the maximum
+	 * of the standard default button width and the required (computed) button width
+	 * for the most wide of the buttons.
+	 * Throws IllegalArgumentException if a button is disposed or does not have GridData.
+	 * 
+	 * @param button
+	 */
+	public static void setEqualButtonWidth(Button... buttons) {
+		if(buttons == null || buttons.length == 0) {
+			return;
+		}
+		int width = convertHorizontalDLUsToPixels(buttons[0], IDialogConstants.BUTTON_WIDTH);
+		for (Button button: buttons) {
+			assertCanSetGridData(button);
+			int w = button.computeSize(SWT.DEFAULT, SWT.DEFAULT).x;
+			if(width < w) {
+				width = w;
+			}
+		}
+		for (Button button: buttons) {
+			((GridData)button.getLayoutData()).widthHint = width;
+		}
+	}
+
+	private static void assertCanSetGridData(Button button) {
+		Assert.isLegal(!DisposeUtils.isDisposed(button));
```

Comment:
Ok, message is added, though it cannot be quite informative as button.getText() throws an exception in this case ;)